internal/app/auth-srv.go

```diff
@@ -0,0 +1,207 @@
+package app
+
+import (
+	"context"
+	"crypto/sha256"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strings"
+
+	"github.com/caos/oidc/pkg/op"
+	"github.com/gorilla/mux"
+	"github.com/labstack/echo/v4"
+	http1 "github.com/reearth/reearth-backend/internal/adapter/http"
+	"github.com/reearth/reearth-backend/internal/app/oauth"
+	"github.com/reearth/reearth-backend/internal/usecase/interactor"
+
+	"github.com/golang/gddo/httputil/header"
+)
+
+func AuthEndPoints(e *echo.Echo, r *echo.Group, cfg *ServerConfig) {
+
+	usersController := http1.NewUserController(interactor.NewUser(cfg.Repos, cfg.Gateways, cfg.Config.SignupSecret))
```

Comment:
This handler behaves like a special controller, so you can use usecase directly instead of using a controller.
```suggestion
	userUsecase :=interactor.NewUser(cfg.Repos, cfg.Gateways, cfg.Config.SignupSecret)
```